pkg/transformer/openshift/openshift.go

```diff
@@ -321,6 +321,10 @@ func (o *OpenShift) Transform(komposeObject kobject.KomposeObject, opt kobject.C
 
 		// Generate pod only and nothing more
 		if service.Restart == "no" || service.Restart == "on-failure" {
+			// Error out if Controller Object is specified with restart: 'on-failure'
+			if opt.IsDeploymentConfigFlag {
+				logrus.Fatalf("Controller object cannot be specified with restart: 'on-failure'")
+			}
```

Comment:
same here.. What happens if the user *doesnt* pass in the flag but still supplies an "on-failure"?